fhd_core/calibration/vis_calibrate.pro

```diff
@@ -57,282 +59,234 @@ FUNCTION vis_calibrate,vis_ptr,cal,obs,status_str,psf,params,jones,vis_weight_pt
-;make sure to plot both, if autocorrelations are used for the calibration solution
-plot_cals,cal,obs,cal_res=cal_res,cal_auto=cal_auto,file_path_base=image_path,_Extra=extra
+  ENDFOR
+  cal_base=Pointer_copy(cal)
 
-if keyword_set(over_calibrate) then begin
-*cal.gain[0] = (*cal_base.gain[0]); + atan(*cal.gain[0],/phase)
-*cal.gain[1] = (*cal_base.gain[1]); + atan(*cal.gain[1],/phase)
-endif
-if keyword_set(perf_calibrate) then begin
-*cal.gain[0] = 1.
-*cal.gain[1] = 1.
-endif
+  ;Perform bandpass (amp and phase per fine freq) and polynomial fitting (low order amp and phase fit plus cable reflection fit)
+  IF Keyword_Set(bandpass_calibrate) THEN BEGIN
+    cal_bandpass=vis_cal_bandpass(cal,obs,params,cal_remainder=cal_remainder,file_path_fhd=file_path_fhd,_Extra=extra)
+    
+    IF Keyword_Set(calibration_polyfit) THEN BEGIN
+        cal_polyfit=vis_cal_polyfit(cal_remainder,obs,_Extra=extra)
+        cal=vis_cal_combine(cal_polyfit,cal_bandpass)
+    ENDIF ELSE cal=cal_bandpass
+    
+  ENDIF ELSE IF Keyword_Set(calibration_polyfit) THEN cal=vis_cal_polyfit(cal,obs,_Extra=extra)
+  
+  ;Get amp from auto-correlation visibilities for plotting (or optionally for the calibration solution itself)
+  IF Keyword_Set(vis_auto_model) THEN cal_auto=vis_cal_auto_fit(obs,cal,vis_auto=vis_auto,vis_model_auto=vis_auto_model,auto_tile_i=auto_tile_i)
+  IF Keyword_Set(calibration_auto_fit) THEN cal_res=vis_cal_subtract(cal_base,cal_auto) ELSE cal_res=vis_cal_subtract(cal_base,cal)
+  
+  basename=file_basename(file_path_fhd)
+  dirpath=file_dirname(file_path_fhd)
+  image_path=filepath(basename,root=dirpath,sub='output_images')
+  ;make sure to plot both, if autocorrelations are used for the calibration solution
+  plot_cals,cal,obs,cal_res=cal_res,cal_auto=cal_auto,file_path_base=image_path,_Extra=extra
 
-if keyword_set(phase_longrun) then begin
-  ;longrun_gain  = getvar_savefile('/nfs/mwa-09/r1/djc/EoR2013/Aug23/fhd_nb_2013longrun/longrun_gain_ave/longrun_gain_plus_phase_dig_poi.sav', 'longrun_gain')
-  longrun_gain = getvar_savefile('/nfs/mwa-09/r1/djc/EoR2013/Aug23/fhd_nb_2013longrun/longrun_gain_ave/longrun_gain_dig_poi_refave_byday.sav','longrun_gain')
-      pointing_num=mwa_get_pointing_number(obs,/string)
-      poi_name = ['-2','-1','0','1','2']
-      poi = where(pointing_num EQ poi_name)
-*cal.gain[0] = abs(*cal.gain[0])*exp(Complex(0,1)*( atan((*cal.gain[0]),/phase)+atan(reform(longrun_gain[0,poi,*,*]),/phase)))
-*cal.gain[1] = abs(*cal.gain[1])*exp(Complex(0,1)*( atan((*cal.gain[1]),/phase)+atan(reform(longrun_gain[1,poi,*,*]),/phase)))
-endif
+  ;In-situ simulation -- forced calibration solutions
+  if keyword_set(sim_over_calibrate) then begin
+    *cal.gain[0] = (*cal_base.gain[0]); + atan(*cal.gain[0],/phase)
+    *cal.gain[1] = (*cal_base.gain[1]); + atan(*cal.gain[1],/phase)
+  endif
+  if keyword_set(sim_perf_calibrate) then begin
+    *cal.gain[0] = 1.
+    *cal.gain[1] = 1.
```

Comment:
Shouldn't this be `cal.gain[0][*] = 1.`? I think this replaces the array of values with a single float.

Reply:
you're right. changed